test/component_test.go

```diff
@@ -0,0 +1,52 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cloudposse/test-helpers/pkg/atmos"
+	helper "github.com/cloudposse/test-helpers/pkg/atmos/component-helper"
+	awshelper "github.com/cloudposse/test-helpers/pkg/aws"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/stretchr/testify/assert"
+)
+
+type ComponentSuite struct {
+	helper.TestSuite
+}
+
+func (s *ComponentSuite) TestBasic() {
+	const component = "ec2-instance/basic"
+	const stack = "default-test"
+	const awsRegion = "us-east-2"
+
+	defer s.DestroyAtmosComponent(s.T(), component, stack, nil)
+	options, _ := s.DeployAtmosComponent(s.T(), component, stack, nil)
+	assert.NotNil(s.T(), options)
+
+	instanceId := atmos.Output(s.T(), options, "instance_id")
+	assert.NotEmpty(s.T(), instanceId)
+
+	privateIp := atmos.OutputList(s.T(), options, "private_ip")
+	assert.EqualValues(s.T(), 1, len(privateIp))
+
+	instanceIds := aws.GetEc2InstanceIdsByTag(s.T(), awsRegion, "size", "xs")
+	instance := awshelper.GetEc2Instances(s.T(), context.Background(), instanceIds[0], awsRegion)
```

Comment:
I think we should go this way as terraform and opentofu tests run in parallel and current implementation could reference wrong instance. 
```suggestion
	instance := awshelper.GetEc2Instances(s.T(), context.Background(), instanceId, awsRegion)
```

Reply:
Could you pls also assert that ec2 instance is running?

Reply:
@goruha Any recommendations or examples of how that is accomplished if Terratest cleans up the instance to a terminated state before the assertions?

Reply:
we terminate instances on component teardown here
https://github.com/cloudposse-terraform-components/aws-ec2-instance/pull/17/files#diff-9d06f2bcfad6ccd9f540153b4993c1fc47dabe60366cae74d02071f5f0b11b16R23

if you want to test that instance removed - you can run
```
s.DestroyAtmosComponent(s.T(), component, stack, nil)

instance := awshelper.GetEc2Instances(s.T(), context.Background(), instanceId, awsRegion)
// Assert that instance stopped 
```

But I think there is no need for such assertions as for terraform cleanup; all resources are managed by it. 
If the defer finished successfully - it's fine

Reply:
@RoseSecurity ^

Reply:
Just fixed it and added that functionality. The IDs output as a list so I had to reference the index of that list and then query for the state!